networkx/utils/backends.py

```diff
@@ -1,104 +1,172 @@
+    - `edge_attrs` : dict, optional
+        Dictionary mapping edge attributes to default values if missing in `G`.
+        If None, then no edge attributes will be converted and default may be 1.
+    - `node_attrs`: dict, optional
+        Dictionary mapping node attributes to default values if missing in `G`.
+        If None, then no node attributes will be converted.
+    - `preserve_edge_attrs` : bool
+        Whether to preserve all edge attributes.
+    - `preserve_node_attrs` : bool
+        Whether to preserve all node attributes.
+    - `preserve_graph_attrs` : bool
+        Whether to preserve all graph attributes.
+    - `preserve_all_attrs` : bool
+        Whether to preserve all graph, node, and edge attributes.
+    - `name` : str
+        The name of the algorithm.
+    - `graph_name` : str
+        The name of the graph argument being converted.
+
+
+Upon setting the appropriate environment variables, NetworkX will:
+
+- Convert NetworkX graphs using `<your_dispatcher_class>.convert_from_nx(G, ...)`.
+- Pass the converted objects to the backend implementation of the algorithm.
+- Convert the result back to a form expected by NetworkX tests using 
+  `<your_dispatcher_class>.convert_to_nx(result, ...)`.
+
+Running Tests
+~~~~~~~~~~~~~~
+
+You can invoke tests using the custom backend with the following command::
+
+    NETWORKX_TEST_BACKEND=<entry_point_name> pytest --pyargs networkx
+
+Notes
+~~~~~~
+
+-   Dispatchable algorithms which are not implemented by the backend
+    will cause a `pytest.xfail()`, giving some indication that not all
+    tests are working, while avoiding causing an explicit failure.
+
+-   If a backend only partially implements some algorithms, it can define
+    a `can_run(name, args, kwargs)` function that returns True or False
+    indicating whether it can run the algorithm with the given arguments.
+
+-   A special `on_start_tests(items)` function may be defined by the backend.
+    It will be called with the list of NetworkX tests discovered. Each item
+    is a test object that can be marked as xfail if the backend does not support
+    the test using `item.add_marker(pytest.mark.xfail(reason=...))`.
 """
+
 import inspect
 import os
-import sys
 import warnings
 from functools import partial
 from importlib.metadata import entry_points
 
 import networkx as nx
 
-from ..exception import NetworkXNotImplemented
-
 __all__ = ["_dispatchable"]
 
 
 def _get_backends(group, *, load_and_call=False):
+    """
+    Retrieve NetworkX `backends` and `backend_info` from the `pyproject.toml` file.
+
+    Parameters
+    -----------
+        group (str): The entry_point to be retrieved.
+        load_and_call (bool, optional): If True, load and call the backend. Defaults to False.
+
+    Returns
+    --------
+        dict: A dictionary mapping backend names to their respective backend objects.
+
+    Raises
+    -------
+        Exception: If an error occurs while loading a backend.
```

Comment:
I'm not seeing where this function will raise `Exception`.  I see where it will issue a warning if `load_and_call` is True and there was an error loading/calling a backend though.

Reply:
Sorry, fixed it.

Thanks!